web/src/components/Header.js

```diff
@@ -53,11 +53,9 @@ class Header extends Component {
         <div className="ds-l-container">
           <div className="ds-l-row">
             <div className="ds-l-col--12 ds-l-md-col--4 site-title">
-              {showSiteTitle && <Link to="/">{t('titleBasic')}</Link>}
-              {!showSiteTitle && !authenticated && (
+              {showSiteTitle || !authenticated ? (
```

Comment:
I had to hop in here when updating the tests for the `Wrapper` component to make sure I understood what it was supposed to do and... this old logic was so hard to make sense of. It covered too many convoluted cases:

- if `showSiteTitle` is true, show the site title
- if `showSiteTitle` is false AND `authenticated` is false, show the site title
- if `showSiteTitle` is false AND `authenticated` is true, show a link to the dashboard

These conditions are exhaustive, so they would be more clearly modeled as an `if/elseif/else`. However, the first two conditions have the same result! We could replace those two conditions with one, and just have an `if/else`:

- if `showSiteTitle` is true OR (`showSiteTitle` is false AND `authenticated` is false), show the site title
- otherwise, show a link to the dashboard

But that looks silly, a true and false on the same variable. That brought back memories of boolean algebra in college, but I couldn't remember all the transitive or associative or whatever else properties. I did remember truth tables, though, so I wrote one of those out, and it ends up that you can simply the above to just this:

- `showSiteTitle` is true OR `authenticated` is false, show the site title
- otherwise, show a link to the dashboard

So yeah, I did all that and turned it into a ternary and hopefully it's easier to read now.

Reply:
![good work](https://media.giphy.com/media/3ohs4pDJmNDDB6NAsM/giphy.gif)

That's some good logic!! Although I did twitch when you mentioned college algebra.